apis/core/v1beta1/featureflag_webhook.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"encoding/json"
+	"fmt"
+
+	_ "embed"
+
+	"github.com/xeipuuv/gojsonschema"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+//go:embed schema/targeting.json
+var TargetingSchema string
+
+//go:embed schema/flags.json
+var FlagsScheme string
+
+// log is for logging in this package.
+var featureflaglog = logf.Log.WithName("featureflag-resource")
+
+func (r *FeatureFlag) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-core-openfeature-dev-v1beta1-featureflag,mutating=false,failurePolicy=fail,sideEffects=None,groups=core.openfeature.dev,resources=featureflags,verbs=create;update,versions=v1beta1,name=vfeatureflag.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &FeatureFlag{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *FeatureFlag) ValidateCreate() error {
+	featureflaglog.Info("validate create", "name", r.Name)
```

Comment:
The error messages are written out on stdout when applying malformed CR, so it would be doubling the output, these logs I find quite informal when debugging if the validation webhook kicked in, they were quite useful for s in the past in other projects

Reply:
I think having log traceability here probably makes sense... it's not like it will hurt performance or anything at the rate we'd expect these to change.

 I will defer to people who know more about debugging operators though.